packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-session/atomic-insight-user-actions-session.tsx

```diff
@@ -0,0 +1,152 @@
+import {Component, h, Prop, Watch, State} from '@stencil/core';
+import {UserAction} from '../..';
+import Flag from '../../../../images/flag.svg';
+import ThreeDotsIcon from '../../../../images/three-dots.svg';
+import {parseTimestampToDateDetails} from '../../../../utils/date-utils';
+import {InitializeBindings} from '../../../../utils/initialization-utils';
+import {InsightBindings} from '../../atomic-insight-interface/atomic-insight-interface';
+import {AtomicInsightUserAction} from '../atomic-insight-user-action/atomic-insight-user-action';
+
+export type UserActionType =
+  | 'SEARCH'
+  | 'CLICK'
+  | 'TICKET_CREATION'
+  | 'VIEW'
+  | 'CUSTOM';
+
+/**
+ * @internal
+ * The `AtomicInsightUserActionSession` component displays all the user actions that took place during a specific user session.
+ * @category Insight Panel
+ * @example
+ * <atomic-insight-user-actions-session userActions={actions} startTimestamp={1723035731}></atomic-insight-user-actions-session>
+ */
+@Component({
+  tag: 'atomic-insight-user-actions-session',
+  styleUrl: 'atomic-insight-user-actions-session.pcss',
+  shadow: true,
+})
+export class AtomicInsightUserActionsSession {
+  @InitializeBindings() public bindings!: InsightBindings;
+  @State() public error!: Error;
+
+  /**
+   * The start time of the session as a Unix timestamp.
+   */
+  @Prop() public startTimestamp!: number;
+  /**
+   * The list of user actions performed during the session.
+   */
+  @Prop() public userActions!: Array<UserAction>;
+
+  private userActionsToDisplay: Array<UserAction> = [];
+  private userActionsAfterCaseCreation: Array<UserAction> = [];
+  @State() areActionsAfterCaseCreationVisible = false;
+
+  connectedCallback() {
+    this.prepareUserActionsToDisplay();
+  }
+
+  @Watch('userActions')
+  prepareUserActionsToDisplay() {
+    const caseCreationIndex = this.userActions.findIndex(
+      ({actionType}) => actionType === 'TICKET_CREATION'
+    );
+    const isCaseCreationSession = caseCreationIndex !== -1;
+
+    if (!isCaseCreationSession) {
```

Comment:
Do you prefer to have the explicit variable?

```suggestion
    if (caseCreationIndex === -1) {
```

Reply:
I do yes